src/lib/components/LoginModal.svelte

```diff
@@ -1,55 +1,40 @@
 <script lang="ts">
 	import { base } from "$app/paths";
 	import { page } from "$app/stores";
-	import { PUBLIC_APP_DATA_SHARING, PUBLIC_APP_NAME, PUBLIC_VERSION } from "$env/static/public";
+	import { PUBLIC_APP_DESCRIPTION, PUBLIC_APP_NAME } from "$env/static/public";
 	import LogoHuggingFaceBorderless from "$lib/components/icons/LogoHuggingFaceBorderless.svelte";
 	import Modal from "$lib/components/Modal.svelte";
 	import type { LayoutData } from "../../routes/$types";
 	import Logo from "./icons/Logo.svelte";
 	export let settings: LayoutData["settings"];
 </script>
 
-<Modal>
+<Modal on:close>
```

Comment:
what does this part do? `on:close` ?

Reply:
It passes the on:close event dispatched from inside `Modal.svelte` to the parent of `LoginModal.svelte` which hides the modal when triggered. See [here](https://github.com/huggingface/chat-ui/blob/e4fc6cab4f542ea187ed91cd651a24c00c6526e7/src/lib/components/chat/ChatWindow.svelte#L54-L59)

tbh I think our modal implementation could be refactored and made a lot simpler but it was a bit out of scope to tackle it here haha